src/contracts/pods/DelayedWithdrawalRouter.sol

```diff
@@ -107,14 +107,39 @@ contract DelayedWithdrawalRouter is Initializable, OwnableUpgradeable, Reentranc
         return _userWithdrawals[user];
     }
 
-    /// @notice Getter function to get all delayedWithdrawals that are currently claimable by the `user`
+    /// @notice Getter function to get all delayedWithdrawals of the `user`
+    /// This function is slightly mis-named as it returns all delayed withdrawals not just claimable ones
     function claimableUserDelayedWithdrawals(address user) external view returns (DelayedWithdrawal[] memory) {
+        uint256 delayedWithdrawalsCompleted = _userWithdrawals[user].delayedWithdrawalsCompleted;
+        uint256 totalDelayedWithdrawals = _userWithdrawals[user].delayedWithdrawals.length;
+        uint256 userDelayedWithdrawalsLength = totalDelayedWithdrawals - delayedWithdrawalsCompleted;
+        DelayedWithdrawal[] memory userDelayedWithdrawals = new DelayedWithdrawal[](userDelayedWithdrawalsLength);
+        for (uint256 i = 0; i < userDelayedWithdrawalsLength; i++) {
+            userDelayedWithdrawals[i] = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+        }
+        return userDelayedWithdrawals;
+    }
+
+    /// @notice Getter function to get all delayedWithdrawals that are currently claimable by the `user`
+    function getClaimableUserDelayedWithdrawals(address user) external view returns (DelayedWithdrawal[] memory) {
         uint256 delayedWithdrawalsCompleted = _userWithdrawals[user].delayedWithdrawalsCompleted;
         uint256 delayedWithdrawalsLength = _userWithdrawals[user].delayedWithdrawals.length;
         uint256 claimableDelayedWithdrawalsLength = delayedWithdrawalsLength - delayedWithdrawalsCompleted;
-        DelayedWithdrawal[] memory claimableDelayedWithdrawals = new DelayedWithdrawal[](claimableDelayedWithdrawalsLength);
+
+        uint256 count = 0;
         for (uint256 i = 0; i < claimableDelayedWithdrawalsLength; i++) {
-            claimableDelayedWithdrawals[i] = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+            DelayedWithdrawal memory delayedWithdrawal = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+            if (block.number > delayedWithdrawal.blockCreated + withdrawalDelayBlocks) {
+                count++;
+            }
+        }
+        
+        DelayedWithdrawal[] memory claimableDelayedWithdrawals = new DelayedWithdrawal[](count);
+        for (uint256 i = 0; i < claimableDelayedWithdrawalsLength; i++) {
+            DelayedWithdrawal memory delayedWithdrawal = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+            if (block.number > delayedWithdrawal.blockCreated + withdrawalDelayBlocks) {
```

Comment:
is there a reason we check this condition in both loops?
this isn't technically wrong + it is 'only a view function' so I'm OK with it, just seems a bit weird / inefficient

Reply:
I think we have to check the if statement in both loops?  Because both loops iterate through the whole list of non completed withdrawals, the first one just finds the length of the array

Reply:
ah I missed this. why aren't we just going through the correct indices in the second loop? this is even more inefficient than I thought.

Reply:
I felt like using a fixed array dynamically wasn't clean but its definitely the more efficient solution, which is more important.  Fixed.